docs/step-by-step/kubeapps-on-tkg/step-2.md

```diff
@@ -0,0 +1,235 @@
+
+1. Declare that the target cluster is using Pinniped by setting the parameter `pinnipedConfig.enable=true`. If using multiple target clusters, please refer to the [Deploying to Multiple Clusters](https://github.com/kubeapps/kubeapps/blob/master/docs/user/deploying-to-multiple-clusters.md) guide. Here is an example:
+
+   > **TIP**: Since the target cluster is the same as the cluster on which Kubeapps is installed, there is no need to set a URL. Note that the `name` field is used only to configure a display name in the Kubeapps dashboard.
+
+   ```yaml
+   clusters:
+     - name: my-tkg-cluster
+       pinnipedConfig:
+         enable: true
+   ```
+
+2. Enable the _Pinniped Proxy_ component so that the requests performed by Kubeapps can be proxied through Pinniped, by setting the parameter `pinnipedProxy.enabled=true`. Here is an example:
+
+   ```yaml
+   pinnipedProxy:
+     enabled: true
+     defaultAuthenticatorName: kubeapps-jwt-authenticator # this name must match the authenticator name previously created
+     defaultPinnipedNamespace: vmware-system-tmc
+     defaultPinnipedAPISuffix: pinniped.tmc.cloud.vmware.com
+   ```
+
+   > **TIP**: The `defaultAuthenticatorName` must match the _JWTAuthenticator_ resource name created in [Step 1](./step-1.md).
+
+3. Configure the _OAuth2Proxy_ component by entering the information gathered from the OIDC provider in [Step 1](./step-1.md). This component performs the authentication flow, generating the appropriate request to the login page and retrieving the token in the callback URL. Here is an example. Remember to replace the placeholders as follows:
+
+   - Replace the `OIDC-ISSUER-URL` with the issuer URL of the OIDC provider. For CSP it is `https://console-stg.cloud.vmware.com/csp/gateway/am/api`.
+   - Replace `CLIENT-ID` with the application ID obtained from the JSON file in the previous step.
+   - Replace `CLIENT-SECRET` with the application secret obtained from the JSON file in the previous step.
+   - Replace `COOKIE-SECRET` with a seed string for secure cookies (should be a 16-, 24-, or 32-byte string).
+
+   ```yaml
+   authProxy:
+     enabled: true
+     provider: oidc
+     clientID: CLIENT-ID
+     clientSecret: CLIENT-SECRET
+     cookieSecret: COOKIE-SECRET
+     additionalFlags:
+       - --oidc-issuer-url=OIDC-ISSUER-URL
+       - --scope=openid email groups
+       - --set-authorization-header=true
+       # - --insecure-oidc-skip-issuer-verification=true
+   ```
+
+   > **NOTE**: In some providers whose issuer URL does not match the token URL (such as VMware CSP), the flag `--insecure-oidc-skip-issuer-verification=true` must be turned on. Be aware of the security concerns of enabling this flag, which are discussed in the [official OAuth2Proxy documentation](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview/).
```

Comment:
```suggestion
   > **NOTE**: In some providers whose issuer URL does not match the token URL (such as VMware CSP), the flag `--insecure-oidc-skip-issuer-verification=true` is currently required. Be aware of the security concerns of enabling this flag, which are discussed in the [official OAuth2Proxy documentation](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview/).
```

Reply:
Have a look at the end of https://github.com/kubeapps/kubeapps/issues/2686#issuecomment-846966217, have you ever tried that manual config instead of relying on the oidc autodiscovery?
If it works, I guess we can send another PR removing all there references to the --insecure flag